### DISCUSSION TITLE
xmake run 没有重新构建

### DISCUSSION THREAD

**Discussion by xinyu391** [xmake-io/xmake#2985] [Q&A]
修改代码后，执行xmake run 新代码没有 编译，run的还是老的程序。
只能 
xmake build 
xmake run 吗？

**Comment by waruqi** [comment]
呢

**Reply by xinyu391** [reply]
原来想提问 xmake run 如何添加程序参数的。
问题写了一半，发现原因了。 xmake run test_demo  sample/abc.txt
程序执行的当前目录不是  xmake 执行的当前目录，而是 test_demo的所在目录导致 sample/abc.txt文件路径不对程序崩溃。
后来改成了 上面这个问题

**Reply by waruqi** [reply]
`xmake run -w /xxx` 或者 `set_rundir`
